src/value/use-template.ts

```diff
@@ -0,0 +1,81 @@
+import { useEffect } from "react"
+import sync from "framesync"
+import { MotionValue } from "."
+import { useMotionValue } from "./use-motion-value"
+
+/**
+ * Combine multiple motion values into a new one using a string template literal.
+ *
+ * ```jsx
+ * import {
+ *   motion,
+ *   useSpring,
+ *   useMotionValue,
+ *   useTemplate
+ * } from "framer-motion"
+ *
+ * function Component() {
+ *   const shadowX = useSpring(0)
+ *   const shadowY = useMotionValue(0)
+ *   const shadow = useTemplate`drop-shadow(${shadowX}px ${shadowY}px 20px rgba(0,0,0,0.3))`
+ *
+ *   return <motion.div style={{ filter: shadow }} />
+ * }
+ * ```
+ *
+ * @public
+ */
+export function useTemplate(
```

Comment:
Is the name `useTemplate` too generic for this purpose? I'm on the fence.

Reply:
Had the same thought. Maybe `useMotionValueTemplate`? 🤔

Reply:
The [original name](https://gist.github.com/InventingWithMonster/38d7cdd3557f398c2ca4a2c81defed08) was `useMotionTemplate` which I feel is a nice mix of specificity and brevity? 

I shortend it because we have a few motion value factories like `useSpring` and `useTransform` without "motion" or "value", but I do agree that out of context, "spring" and to a lesser extend "transform" more clearly live in an animation-esque context than "value" and "template"

Reply:
Let's go with useMotionTemplate, then!